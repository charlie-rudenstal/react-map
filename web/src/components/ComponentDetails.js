import React from 'react';
import provideApi from '../lib/provideApi';
import styles from './ComponentDetails.less';

export function ComponentDetails({component}) {
  if (!component) return <div>Fetching component data...</div>;
  return (
    <div className={styles.content}>
      <section className="paper">
        <h2 className="paper__title">Preview</h2>
        <iframe src={`/render/component/${component.path}`} style={{ border: 'none' }} />
      </section>

      <section className="paper">
        <h2 className="paper__title">Children</h2>
        <div>
          {component.children.map(item => (
            <div className="tag" key={item.name}>{item.name}</div>
          ))}
        </div>
      </section>

      <section className="paper">
        <h2 className="paper__title">Class names</h2>
        <div>
          {component.classNames.map(item => (
            <div className="tag" key={item.name}>{item.name}</div>
          ))}
        </div>
      </section>

      <section className="paper">
        <h2 className="paper__title">Dependencies</h2>
        <div>
          {component.dependencies.map(item => (
            <div className="tag" key={item.name}>{item.name} ({item.path})</div>
          ))}
        </div>
      </section>
    </div>
  );
}

ComponentDetails.propTypes = {
  path: React.PropTypes.string.isRequired
};

export default provideApi(ComponentDetails,
  props => `/component/${props.path}`,
  component => ({ component })
);
