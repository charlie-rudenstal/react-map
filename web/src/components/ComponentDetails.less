:local(.content) {
  margin: 12px;
}