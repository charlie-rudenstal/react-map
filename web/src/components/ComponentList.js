import React from 'react';

class ComponentList extends React.Component {

  constructor(props) {
    super(props);
    this.state = {components: []};
  }

  componentDidMount() {
    fetch('/api/components')
      .then(res => res.json())
      .then(({components}) => {
        this.setState({components});
      });
  }

  render() {
    return (
      <div>
        <div>ComponentList</div>
        <ul>
          {this.state.components.map(component => (
            <div key={component.name}>
              {component.name}
            </div>
          ))}
        </ul>
      </div>
    );
  }

}

export default ComponentList;