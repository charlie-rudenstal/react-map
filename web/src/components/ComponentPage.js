import React from 'react';

function ComponentPage() {
  return (
    <div>Component Page</div>
  );
}

export default ComponentPage;