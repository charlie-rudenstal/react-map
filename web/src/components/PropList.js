import React from 'react';
import provideApi from '../lib/provideApi';
import { withRouter } from 'react-router';

function PropList({component, router}) {
  if (!component) return null;
  console.log('component is', component);
  return (
    <div className="itemlist">
      {component.props.map(({name, type}) => (
        <div className="itemlist__item">
          {name}
        </div>
      ))}
    </div>
  );
}

export default withRouter(provideApi(PropList,
  props => `/component/${props.router.params.splat}`,
  component => ({ component })
));
