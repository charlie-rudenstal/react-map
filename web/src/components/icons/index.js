export {default as FileIcon} from './File';
