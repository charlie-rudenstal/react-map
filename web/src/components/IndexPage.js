import React from 'react';
import ComponentList from './ComponentList';

function IndexPage() {
  return (
    <div>
      <ComponentList />
    </div>
  );
}

export default IndexPage;
