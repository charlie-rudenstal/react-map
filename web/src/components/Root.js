import React from 'react';
import ComponentList from './ComponentList';

function Root() {
  return (
    <div>
      <ComponentList />
    </div>
  );
}

export default Root;
