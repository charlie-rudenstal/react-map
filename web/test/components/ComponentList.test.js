import React from 'react';
import renderer from 'react-test-renderer';
import { ComponentList } from '../../src/components/ComponentList';

describe('ComponentList', () => {
  it('should render correctly', () => {
    const components = [
      { name: "Button", path: "Button.js" },
      { name: "Tab", path: "Tab.js" },
      { name: "Tabs", path: "Tabs.js" }
    ];

    const tree = renderer.create(<ComponentList components={components} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
