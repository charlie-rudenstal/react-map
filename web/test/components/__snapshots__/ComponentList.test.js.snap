exports[`ComponentList should render correctly 1`] = `
<nav
  className="itemlist">
  <a
    className="itemlist__item"
    onClick={[Function]}
    style={Object {}}>
    Button
  </a>
  <a
    className="itemlist__item"
    onClick={[Function]}
    style={Object {}}>
    Tab
  </a>
  <a
    className="itemlist__item"
    onClick={[Function]}
    style={Object {}}>
    Tabs
  </a>
</nav>
`;
